test/tests/selftest/pbs_manager.py

```diff
@@ -0,0 +1,63 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.selftest import *
+from ptl.utils.pbs_dshutils import PbsConfigError
+
+
+class TestManager(TestSelf):
+    """
+    Test ptl.server.manager()
+    """
+
+    def test_pbs_conf(self):
+        """
+        Test setting an string array with quote characters
+        """
+        attr = {}
+        attr['type'] = 'string_array'
+        attr['flag'] = 'h'
+        r = 'foo2'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+        self.assertEqual(rc, 0)
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {"resources_available.foo2": ["A'A", 'B"B', 'C C']})
+        nodes = self.server.filter(SERVER,
+                                   {'resources_available.foo2':
+                                    """A'A,B"B,C C"""})
+        self.logger.info(nodes)
```

Comment:
You're querying the server, nodes is a poor variable name.  Also, don't log a blanket variable.  Either turn it into a full log message, or maybe assert instead.